File: pgm/ABC_Dart/ABC_249_A.dart
// import 'dart:collection';
// import 'dart:convert';
import 'dart:io';
import 'dart:math';
// import 'dart:typed_data';

void main() {
  final [int A, int B, int C, int D, int E, int F, int X] = stdin
      .readLineSync()!
      .split(' ')
      .map(int.parse)
      .toList();
  const String Takahashi = "Takahashi";
  const String Aoki = "Aoki";
  const String Draw = "Draw";
  int takahashiDistance = B * A * (X ~/ (A + C)) + B * min(A, X % (A + C));
  int aokiDistance = E * D * (X ~/ (D + F)) + E * min(D, X % (D + F));
  if (takahashiDistance > aokiDistance) {
    print(Takahashi);
  } else if (takahashiDistance < aokiDistance) {
    print(Aoki);
  } else {
    print(Draw);
  }
  exit(0);
}

File: pgm/ABC_Kotlin/ABC_249_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.LocalTime
// import java.time.format.DateTimeFormatter
import java.util.Scanner
// import java.util.StringTokenizer

import kotlin.math.*

const val Takahashi: String = "Takahashi"
const val Aoki: String = "Aoki"
const val Draw: String = "Draw"

fun main() {
    val scanner: Scanner = Scanner(System.`in`)
    val A: Int = scanner.nextInt()
    val B: Int = scanner.nextInt()
    val C: Int = scanner.nextInt()
    val D: Int = scanner.nextInt()
    val E: Int = scanner.nextInt()
    val F: Int = scanner.nextInt()
    val X: Int = scanner.nextInt()
    var takahashiDistance: Int = B * A * (X / (A + C)) + B * min(A, X % (A + C))
    var aokiDistance: Int = E * D * (X / (D + F)) + E * min(D, X % (D + F))
    if (takahashiDistance > aokiDistance) {
        println(Takahashi)
    } else if (takahashiDistance < aokiDistance) {
        println(Aoki)
    } else {
        println(Draw)
    }
    kotlin.system.exitProcess(0)
}